File: android/src/main/java/com/golfchannel/reactnative/marketingcloud/RNMarketingCloudConfig.kt
package com.golfchannel.reactnative.marketingcloud

import android.content.Intent
import androidx.annotation.DrawableRes
import com.salesforce.marketingcloud.notifications.NotificationMessage

class RNMarketingCloudConfig {
  lateinit var appId: String
  lateinit var accessToken: String
  lateinit var cloudServerURL: String
  lateinit var senderId: String

  @JvmField @DrawableRes
  var notificationIcon: Int = 0
  lateinit var notificationUrlHandler: (url: String?) -> Intent

  @JvmField
  var analyticsEnabled = false

  @JvmField
  var inboxEnabled = false
}

File: ios/RNMarketingCloudConfig.swift
import Foundation

@objc public class RNMarketingCloudConfig: NSObject {
  @objc public var appId: String?
  @objc public var accessToken: String?
  @objc public var cloudServerURL: URL?
  @objc public var mid: String?

  @objc public var analyticsEnabled = false
  @objc public var applicationControlsBadging = false
  @objc public var locationEnabled = false
  @objc public var inboxEnabled = false
}
